### PR TITLE
Flavor of iran added and tested

### PR DESCRIPTION
**Pull request by ah8ad3**
hi,
i have pushed this that include iran Postal code and Id card validation plus province select
also i open an issue  #358 last night about it
also i don't know why norway tests not pass so i fix them too
@claudep

### CODE REVIEW

**Comment by benkonrath** [comment]
I've fixed the issues with the Norway localflavor in master as well. Can you rebase this PR with master? I'll do the review after the rebase. Thanks.

**Comment by ah8ad3** [comment]
Hi @benkonrath 
i update project with master and it's ok

File: docs/changelog.rst [hunk 1]
```diff
@@ -6,15 +6,15 @@ Changelog
...
+- Added local flavor for Iran
 
 New fields for existing flavors:
 
 - None
 
 Modifications to existing flavors:
 
-- None
+- Improve prospector check for norway in `localflavor/no/forms.py` line 132 and 147
```
Comment (benkonrath) [inline docs/changelog.rst]: This should be removed now that the fix isn't in this PR.

**Comment by ah8ad3** [comment]
I've fixed  the changelog.rst, also i rebase project with latest update
@benkonrath

File: localflavor/ir/forms.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            return ''
+
+        match = id_number_re.match(value)
+        if not match:
+            raise ValidationError(self.error_messages['invalid'])
+
+        check = int(value[9])
+        s = sum([int(value[x]) * (10 - x) for x in range(9)]) % 11
+
+        if (2 > s == check) or (s >= 2 and check + s == 11):
```
Comment (benkonrath) [inline localflavor/ir/forms.py]: I don't really follow this part of the if statement: `(2 > s == check)` it looks like there's a comparison between a number and boolean. Is intentional?

File: localflavor/ir/forms.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            return value
+        else:
+            raise ValidationError(self.error_messages['invalid'])
```
Comment (benkonrath) [inline localflavor/ir/forms.py]: There isn't a test for this validation error path. Is it possible to add one?

File: localflavor/ir/forms.py (hunk 2 above)
Comment (ah8ad3) [reply localflavor/ir/forms.py]: this is the same 
`2 > s and s == check`

File: localflavor/ir/forms.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        if value not in self.empty_values:
+            value = value.replace(' ', '')
+        return super(IRPostalCodeField, self).clean(value)
+
+
+class IRIDNumberField(Field):
+    """
+    A form field that validates its input as an Iranian identification number.
+
+    Valid form is per the Iranian ID specification.
```
[the remark this answers is not in the document]
Comment (ah8ad3) [reply localflavor/ir/forms.py]: i can't find good English doc so i add Persian and suggest to translate it with google translate

File: localflavor/ir/forms.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    default_error_messages = {
+        'invalid': _('Enter a valid ID number.'),
+    }
+
+    def clean(self, value):
+        value = super(IRIDNumberField, self).clean(value)
+
+        if value in EMPTY_VALUES:
+            return ''
```
[the remark this answers is not in the document]
Comment (ah8ad3) [reply localflavor/ir/forms.py]: good one, i fix that

File: localflavor/ir/forms.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+from django.forms.fields import Field, RegexField, Select
+from django.utils.translation import ugettext_lazy as _
+
+from .ir_provinces import PROVINCE_CHOICES
+
+id_number_re = re.compile(r'^\d{10}$')
+
+
+class IRProvinceSelect(Select):
+    """A Select widget that uses a list of Iran provinces cities as its choices."""
```
[the remark this answers is not in the document]
Comment (ah8ad3) [reply localflavor/ir/forms.py]: actually at first i don't know where i must add this `.. versionadded:: 2.2` tags,
i fix that

File: localflavor/ir/forms.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+import re
+
+from django.core.exceptions import ValidationError
+from django.core.validators import EMPTY_VALUES
+from django.forms.fields import Field, RegexField, Select
+from django.utils.translation import ugettext_lazy as _
+
+from .ir_provinces import PROVINCE_CHOICES
+
+id_number_re = re.compile(r'^\d{10}$')
```
[the remark this answers is not in the document]
Comment (ah8ad3) [reply localflavor/ir/forms.py]: good point, i fix that

File: localflavor/ir/forms.py (hunk 3 above)
Comment (ah8ad3) [reply localflavor/ir/forms.py]: i miss this one,
i added the test

**Comment by ah8ad3** [comment]
hi @benkonrath 
i think i fix all the request changes

File: localflavor/ir/forms.py (hunk 2 above)
Comment (benkonrath) [reply localflavor/ir/forms.py]: Ah ok, so it's just a shortcut that I didn't know. Thanks.

File: localflavor/ir/forms.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+class IRIDNumberField(Field):
+    """
+    A form field that validates its input as an Iranian identification number.
+
+    Valid form is per the Iranian ID specification.
+
+    persian documentation :
+        http://www.aliarash.com/article/codemeli/codemeli.htm
+
+        i can't find good english doc so i propose to use google translate for this one
```
Comment (benkonrath) [inline localflavor/ir/forms.py]: I think we can be a little more clear with this comment:
```suggestion
        There isn't good English documentation available for the Iranian identification number. Non-Persian speakers
        will need to use an online translation service to read this documentation.
```

File: localflavor/ir/forms.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        return super(IRPostalCodeField, self).clean(value)
+
+
+class IRIDNumberField(Field):
+    """
+    A form field that validates its input as an Iranian identification number.
+
+    Valid form is per the Iranian ID specification.
+
+    persian documentation :
```
Comment (benkonrath) [inline localflavor/ir/forms.py]: Small tweak: 
```suggestion
    Persian documentation :
```

**Comment by benkonrath** [comment]
Looks good. Thanks for the explanation and the updates. We just need those two small comment updates that I suggested and I'll merge it. Thanks!

**Pull request merged**